pkg/cmd/cli/cmd/set/env.go

```diff
@@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
 	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
 	outputFormat := kcmdutil.GetFlagString(cmd, "output")
+	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
```

Comment:
Volume mount knobs is "oc set volume", not "set env".

I'm trying to imagine how someone would craft the "map key -> name" setup
in a non-confusing manner.

On Tue, Jun 7, 2016 at 9:54 PM, Jordan Liggitt <notifications@github.com>
wrote:

> In pkg/cmd/cli/cmd/set/env.go
> <https://github.com/openshift/origin/pull/9210#discussion_r66182489>:
>
> > @@ -119,6 +128,9 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
> >  	//overwrite := kcmdutil.GetFlagBool(cmd, "overwrite")
> >  	resourceVersion := kcmdutil.GetFlagString(cmd, "resource-version")
> >  	outputFormat := kcmdutil.GetFlagString(cmd, "output")
> > +	secretName := kcmdutil.GetFlagString(cmd, "from-secret")
>
> which keys and what envvars names they map to both seem like something
> we'd want, especially if we're pulling info from secrets/configmaps we
> don't control (as in the case of service linking) to provide envvars to
> containers we also may not control
>
> that info seems like a subset of secret/configmap volume mount knobs we
> want (add in file modes and owner info for that)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9210/files/2dca495d9d658651af8fa61da45494ba31c65f6f#r66182489>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p6bSr8aDMdO7dX7-qdqlitV3ggFCks5qJiDIgaJpZM4IwT6g>
> .
>